src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -138,10 +137,11 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
         private async Task<AuthenticationResult> RefreshRtOrFetchNewAccessTokenAsync(CancellationToken cancellationToken)
         {
             var logger = AuthenticationRequestParameters.RequestContext.Logger;
-            
-            if (ApiEvent.IsLongRunningObo(AuthenticationRequestParameters.ApiId))
+
+            // InitiateLongRunningProcessInWebApi retrieves tokens only with assertion, not by refresh token

```

Comment:
The reason why we don't use RT here is because of #3825. If tokens are revoked in AAD, refreshing with this cached RT will throw an AAD exception. Since we already have a user assertion, use it to fetch new tokens via OBO grant. The expectation is that this use assertion is newer than the one that was used to acquire cached RT.

Reply:
Pls add this comment in the code.